### PR TITLE
Fix ini_parse_quantity()

### LINKED ISSUE #448: Backport fix for ini_parse_quantity from PHP
Tests are currently failing, likely related to https://github.com/php/php-src/pull/11910  and https://github.com/php/php-src/issues/11876

@fisharebest would you like to have a look and submit a fix?

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Fix #448

/cc @fisharebest could you please have a look?
@TimWolla also maybe?

### CODE REVIEW

File: src/Php82/Php82.php [hunk 1]
```diff
@@ -174,6 +174,42 @@ public static function ini_parse_quantity(string $value): int
...
+                ++$digits_consumed;
+            }
+            if ($digits_consumed !== $str_end && ($value[$digits_consumed] === '+' || $value[$digits_consumed] === '-')) {
+                ++$digits_consumed;
+            }
+
+            if ($value[$digits_consumed] === '0') {
+                /* Value is just 0 */
+                if ($digits_consumed + 1 === $str_end) {
+                    return 0;
```
Comment (TimWolla) [inline src/Php82/Php82.php]: I believe `goto evaluation;` or actually adding the same helper function `zend_ini_consume_quantity_prefix` would match the “style” of the original implementation better.

But trying to understand the code, I realize that the original implementation is needlessly complicated / still allows for funky stuff like `0x+0` and `0x     0`. /cc @girgias

Comment (Girgias) [reply src/Php82/Php82.php]: I don't think it allows a second + or whitespace, this is why we needed to use two loops to handle the prefix and suffix shenanigans. But I might have implemented this wrong, I do know there are some weird cases that are still possible tho :-/

**Comment by fisharebest** [comment]
@nicolas-grekas - the logic seems sound, and it passes my "brute-force" test, so it looks good to me.

File: src/Php82/Php82.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Php82/Php82.php]: > goto evaluation

updated!

File: src/Php82/Php82.php [hunk 2]
```diff
@@ -197,16 +233,6 @@ public static function ini_parse_quantity(string $value): int
 
         $digits_end = $digits;
 
-        // The native function treats 0x0x123 the same as 0x123.  This is a bug which we must replicate.
```
Comment (nicolas-grekas) [inline src/Php82/Php82.php:200]: I just removed this block /cc @fisharebest 
Nothing else to cleanup AFAYK?

**Pull request merged**